digdag-standards/src/main/java/io/digdag/standards/operator/RbOperatorFactory.java

```diff
@@ -158,6 +160,20 @@ private Config runCode(Config params)
             int ecode = p.waitFor();
 
             if (ecode != 0) {
+                // If a ruby error message and stacktrace are available in outFile, throw it.
+                // Otherwise, fallback to RuntimeException.
+                try {
+                    Config out = mapper.readValue(workspace.getFile(outFile), Config.class);
+                    Config err = out.getNestedOrGetEmpty("error");
+                    Optional<String> message = err.getOptional("message", String.class);
+                    Optional<String> stacktrace = err.getOptional("stacktrace", String.class);
+                    if (message.isPresent() && stacktrace.isPresent()) {
+                        throw new RbRuntimeException(message.get(), stacktrace.get());
```

Comment:
Do you have any strong reasons or concerns why you tried to define `CommandRuntimeException` and `RbRuntimeException`? I think that it's good for us to use `RuntimeException` and enhance the message as the 1st argument like following. The current message `"Ruby command failed with code.."` has not been helpful so much for users. I also don't think that it's good idea to handles `CommandRuntimeException` in `OperatorManager` because of layer violation between operator and the manager.

```java
    String reason = "Ruby command failed with code " + ecode +
        (message.isPresent() ? "\n..." : "") +
        (stacktrace.isPresent() ? "\n..." : "");
```

What do you think?

Reply:
I felt the current (I mean that, before this PR) stacktrace, which is like below, is useless.
So, I replaced it, too. Do you think the Java stacktrace is useful?

```
io.digdag.standards.operator.RbOperatorFactory$RbOperator.runCode(RbOperatorFactory.java:161),
io.digdag.standards.operator.RbOperatorFactory$RbOperator.runTask(RbOperatorFactory.java:92), 
io.digdag.util.BaseOperator.run(BaseOperator.java:35), 
io.digdag.core.agent.OperatorManager.callExecutor(OperatorManager.java:312), 
io.digdag.core.agent.OperatorManager.runWithWorkspace(OperatorManager.java:254), 
io.digdag.core.agent.OperatorManager.lambda$runWithHeartbeat$2(OperatorManager.java:137), 
io.digdag.core.agent.ExtractArchiveWorkspaceManager.withExtractedArchive(ExtractArchiveWorkspaceManager.java:77), 
io.digdag.core.agent.OperatorManager.runWithHeartbeat(OperatorManager.java:135), 
io.digdag.core.agent.OperatorManager.run(OperatorManager.java:119), 
io.digdag.core.agent.MultiThreadAgent.lambda$null$0(MultiThreadAgent.java:127), 
java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511), 
java.util.concurrent.FutureTask.run(FutureTask.java:266), 
java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149), 
java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624), 
java.lang.Thread.run(Thread.java:748)
```

Reply:
I will add tests after specification is fixed.

Reply:
@muga

Reply:
@muga any updates about this?

Reply:
@sonots I will update within this week. Sorry to late response.

Reply:
@muga could you fix specification?

Reply:
@muga could you fix specification?

Reply:
No discussion happened.

Since I can't have plenty of time working for digdag anymore, I updated not to define RbRuntimeException and simply define more elaborated ruby message and throw RuntimeException.